src/rgw/rgw_redis_lock.cc

```diff
@@ -0,0 +1,130 @@
+#include "rgw_redis_lock.h"
+
+#include <boost/asio/detached.hpp>
+#include <fstream>
+#include <iostream>
+#include <string>
+
+#include "common/async/blocked_completion.h"
+
+namespace rgw {
+namespace redislock {
+
+struct initiate_exec {
+  connection* conn;
+
+  using executor_type = boost::redis::connection::executor_type;
+  executor_type get_executor() const noexcept { return conn->get_executor(); }
+
+  template <typename Handler, typename Response>
+  void operator()(Handler handler, const boost::redis::request& req,
+                  Response& resp) {
+    auto h = boost::asio::consign(std::move(handler), conn);
+    return boost::asio::dispatch(
+        get_executor(), [c = conn, &req, &resp, h = std::move(h)]() mutable {
+          return c->async_exec(req, resp, std::move(h));
+        });
+  }
+};
+
+template <typename Response, typename CompletionToken>
+auto async_exec(connection* conn, const boost::redis::request& req,
+                Response& resp, CompletionToken&& token) {
+  return boost::asio::async_initiate<
+      CompletionToken, void(boost::system::error_code, std::size_t)>(
+      initiate_exec{conn}, token, req, resp);
+}
+
+template <typename T>
+void redis_exec(connection* conn, boost::system::error_code& ec,
+                boost::redis::request& req, boost::redis::response<T>& resp,
+                optional_yield y) {
+  if (y) {
+    auto yield = y.get_yield_context();
+    async_exec(conn, req, resp, yield[ec]);
+  } else {
+    async_exec(conn, req, resp, ceph::async::use_blocked[ec]);
+  }
+}
+
+template <typename T>
+int doRedisFunc(connection* conn, boost::redis::request& req,
+                boost::redis::response<T>& resp, optional_yield y) {
+  boost::system::error_code ec;
+  redis_exec(conn, ec, req, resp, y);
+
+  if (ec) {
+    std::cerr << "EC Message: " << ec.message() << std::endl;
+    return ec.value();
+  }
+  return std::get<0>(resp).value();
+}
+
+// TODO: Remove this later
+std::string getLuaScript(const std::string& filename) {
+  std::ifstream file(filename);
+  if (!file.is_open()) {
+    std::cerr << "Could not open the file: " << filename << std::endl;
+    std::cerr << file.rdstate() << std::endl;
+    return "";
+  }
+  std::stringstream buffer;
+  buffer << file.rdbuf();
+
+  std::string luaScript = buffer.str();
+  file.close();
+  return luaScript;
+}
+
+int initLock(boost::asio::io_context& io, connection* conn, config* cfg,
```

Comment:
please rename to `connect_with_lock`

Reply:
this will also allow adding support for multiple functionalities (locks, queue etc.) on the same connection.
for that, we would probbaly to use "rgwlib" as the libarary name, as we want different functionalities to be able to add multiple functions to the same lib

Reply:
there is already an an object using that name, and there's a librgw.so;  what kind of library are we building?

Reply:
not a C/C++ library.
this is a set of lua scripts we load into the redis server, that are later used by the RGW.
this is similar to the way we load object classes into the OSD. 
one difference is that redis supports only one lua module to hold all scripts, so we have to collect all the scripts that we need (lock, queue etc.) into one module that we call "rgwlib"

Reply:
please rename, the function as this does not init any lock. just add lock support to redis

Reply:
We actually do not need the `initLock/initQueue` anymore. Since they load the same script. 
We can directly call the `rgw::redis::load_lua_rgwlib`. (Open to suggestions for other names)

Do you think it helps to still have a separate abstraction on top of this?

Reply:
lets leave that out fro a lock.
so, the lock interface does not have a init/remove API